src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,96 @@
+package duke;
+
+import java.io.IOException;
+
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Duke class that initialises the duke.Duke chat bot.
+ * The Duke class supports operators including
+ * (i) run: runs the chat bot
+ */
+
+public class Duke {
+
+    /** Path to data storage file. */
+    private static final String LOCATION_OF_FILE = "data.txt";
+
+    /** Handles loading and saving of Tasks. */
+    private final Storage storage;
+
+    /** Stores all tasks */
+    private TaskList tasks;
+
+    /** Handles interactions with users. */
+    private final Ui ui;
+
+    /** Parses commands by users for Duke to understand. */
+    private final Parser parser;
+
+    /**
+     * Constructor for the Duke chatbot.
+     * Loads any pre-existing data from the provided filePath.
+     */
+
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage(LOCATION_OF_FILE);
+
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (IOException | DukeException e) {
+            tasks = new TaskList();
+            ui.printMessage(e.getMessage());
+        } finally {
+            parser = new Parser(tasks, ui, storage);
+        }
+    }
+
+    /**
+     * Runs the command line version of the Duke chat bot.
+     */
+
+    private void run() {
+
+        String input;
+
+        ui.printMessage(Ui.greet());
```

Comment:
Maybe consider `ui.greet()` internally print the greeting message?

Rerouting ui's return value back to ui seems to break OOP's principle of encapsulation